processing/src/main/java/org/apache/druid/segment/virtual/ExpressionSelectors.java

```diff
@@ -498,7 +503,7 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
    */
   private static Object coerceListDimToStringArray(List val)
   {
-    Object[] arrayVal = val.stream().map(Object::toString).toArray(String[]::new);
+    Object[] arrayVal = val.stream().map(x -> x != null ? x.toString() : x).toArray(String[]::new);
```

Comment:
nit: `x != null ? x.toString() : null` looks more clear to me.

Reply:
I have some more follow-up work to do, will try to revisit and clean stuff up a bit more in the future.